toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -114,16 +130,73 @@ func (b *BootCustomizer) getSELinuxModeFromGrub() (imagecustomizerapi.SELinuxMod
 func (b *BootCustomizer) GetSELinuxMode(imageChroot safechroot.ChrootInterface) (imagecustomizerapi.SELinuxMode, error) {
 	// Get the SELinux mode from the kernel command-line args.
 	selinuxMode, err := b.getSELinuxModeFromGrub()
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return imagecustomizerapi.SELinuxModeDefault, err
+	} else if !errors.Is(err, fs.ErrNotExist) && selinuxMode != imagecustomizerapi.SELinuxModeDefault {
+		return selinuxMode, nil
+	}
+
+	// Fallback to extracting from UKI if grub.cfg doesn't exist or returns default.
+	selinuxMode, err = b.getSELinuxModeFromUki(imageChroot)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return imagecustomizerapi.SELinuxModeDefault, fmt.Errorf("failed to extract SELinux mode from UKI:\n%w", err)
+	} else if !errors.Is(err, fs.ErrNotExist) && selinuxMode != imagecustomizerapi.SELinuxModeDefault {
+		return selinuxMode, nil
+	}
+
+	// Final fallback: Get the SELinux mode from the /etc/selinux/config file.
+	selinuxMode, err = getSELinuxModeFromConfigFile(imageChroot)
 	if err != nil {
 		return imagecustomizerapi.SELinuxModeDefault, err
 	}
 
-	if selinuxMode == imagecustomizerapi.SELinuxModeDefault {
-		// Get the SELinux mode from the /etc/selinux/config file.
-		selinuxMode, err = getSELinuxModeFromConfigFile(imageChroot)
-		if err != nil {
-			return imagecustomizerapi.SELinuxModeDefault, err
-		}
+	return selinuxMode, nil
+}
+
+// Extract SELinux mode from UKI kernel command-line arguments.
+func (b *BootCustomizer) getSELinuxModeFromUki(imageChroot safechroot.ChrootInterface) (imagecustomizerapi.SELinuxMode, error) {
+	espDir := filepath.Join(imageChroot.RootDir(), EspDir)
+	buildDir := filepath.Join(filepath.Dir(imageChroot.RootDir()), "uki-selinux-temp")
+
+	err := os.MkdirAll(buildDir, 0o755)
+	if err != nil {
+		return imagecustomizerapi.SELinuxModeDefault, fmt.Errorf("failed to create temp directory: %w", err)
+	}
+	defer os.RemoveAll(buildDir)
+
+	kernelToArgs, err := extractKernelCmdlineFromUkiEfis(espDir, buildDir)
+	if err != nil {
+		return imagecustomizerapi.SELinuxModeDefault, err
+	}
+
+	if len(kernelToArgs) == 0 {
+		return imagecustomizerapi.SELinuxModeDefault, fs.ErrNotExist
+	}
+
+	// Use the first kernel's cmdline since they should all have the same SELinux settings.
+	var firstCmdline string
```

Comment:
De-duplicate code between `getSELinuxModeFromUki` and `extractKernelCmdlineFromUki`.